sympy/simplify/hyperexpand.py

```diff
@@ -2332,8 +2364,10 @@ def do_slater(an, bm, ap, bq, z, zfinal):
                 s = Dummy('s')
                 integrand = z**s
                 for b in bm:
-                    if not Mod(b, 1) and b.is_Number:
-                        b = int(round(b))
+                    if b.is_Number:  # XXX is it necessary to convert to int?
```

Comment:
gamma function does not like floating point numbers that are integers in the sense that Mod(x, 1) is zero. So those should be converted to (first int's, and then at some point) Integers. Other numbers should not be converted so the Mod(b, 1) is needed.